src/charm_state.py

```diff
@@ -114,7 +240,109 @@ def from_charm(cls, charm: CharmBase) -> "CharmConfig":
         else:
             openstack_clouds_yaml = None
 
-        return cls(runner_storage=runner_storage, openstack_clouds_yaml=openstack_clouds_yaml)
+        return cls(
+            path=path,
+            token=token,
+            reconcile_interval=reconcile_interval,
+            denylist=denylist,
+            dockerhub_mirror=dockerhub_mirror,
+            openstack_clouds_yaml=openstack_clouds_yaml,
+        )
+
+    @root_validator
+    @classmethod
+    def check_fields(cls, values: dict) -> dict:
+        """Validate the general charm configuration."""
+        reconcile_interval = values.get("reconcile_interval")
+        # By property definition, this cannot be None. Needed for mypy type check.
+        assert reconcile_interval is not None, "Unreachable code"  # nosec for [B101:assert_used]
+
+        if reconcile_interval < 2:
```

Comment:
Why 2, is it a heuristic? This should also be added to the config description.

Reply:
The reason is `EventTimer` sets a timeout of `reocncile_interval - 1` mintues. Therefore, the interval must be at least 2 or the timeout will be 0 mins or less.

This configuration is generally needs to be much larger than 2 minutes. So I am unsure if we need to document this in the `config.yaml`.